Doesn't build on Manjaro ARM

I was able to build kanata on regular x86_64 PC without any issues, but it fails to build on Pinebook Pro under Manjaro ARM. Error is

```
   Compiling kanata v1.0.1 (/home/fenuks/Projects/keyboard/kanata)
error[E0308]: mismatched types
  --> src/oskbd/linux.rs:91:29
   |
91 |             uidev.name[0] = 'k' as i8;
   |             -------------   ^^^^^^^^^ expected `u8`, found `i8`
   |             |
   |             expected due to the type of this binding

error[E0308]: mismatched types
  --> src/oskbd/linux.rs:92:29
   |
92 |             uidev.name[1] = 't' as i8;
   |             -------------   ^^^^^^^^^ expected `u8`, found `i8`
   |             |
   |             expected due to the type of this binding

error[E0308]: mismatched types
  --> src/oskbd/linux.rs:93:29
   |
93 |             uidev.name[2] = 'r' as i8;
   |             -------------   ^^^^^^^^^ expected `u8`, found `i8`
   |             |
   |             expected due to the type of this binding

error[E0308]: mismatched types
  --> src/oskbd/linux.rs:94:29
   |
94 |             uidev.name[3] = 'l' as i8;
   |             -------------   ^^^^^^^^^ expected `u8`, found `i8`
   |             |
   |             expected due to the type of this binding

For more information about this error, try `rustc --explain E0308`.
error: could not compile `kanata` due to 4 previous errors
```

Changing these casts to u8 fixes compilation error, so it seems it's portability problem, and c_char on my CPU is u8, as seen in libc sources:

```
//! AArch64-specific definitions for 64-bit linux-like values

pub type c_char = u8;
```

I'm not that familiar with rust, but perhaps code should be changed to something to this in order to fix problem for various strange machines:

```diff
diff --git i/src/oskbd/linux.rs w/src/oskbd/linux.rs
index a7de646..b3424ae 100644
--- i/src/oskbd/linux.rs
+++ w/src/oskbd/linux.rs
@@ -14,6 +14,7 @@
 use crate::custom_action::*;
 use crate::keys::*;
 use libc::input_event as raw_event;
+use libc::c_char;
 
 // file i/o
 use io::Write;
@@ -88,10 +89,10 @@ pub fn new() -> Result<Self, io::Error> {
             }
 
             let mut uidev: uinput_user_dev = mem::zeroed();
-            uidev.name[0] = 'k' as i8;
-            uidev.name[1] = 't' as i8;
-            uidev.name[2] = 'r' as i8;
-            uidev.name[3] = 'l' as i8;
+            uidev.name[0] = 'k' as c_char;
+            uidev.name[1] = 't' as c_char;
+            uidev.name[2] = 'r' as c_char;
+            uidev.name[3] = 'l' as c_char;
             uidev.id.bustype = 0x3; // BUS_USB
             uidev.id.vendor = 0x1;
             uidev.id.product = 0x1;
```

--- a/src/oskbd/linux.rs
+++ b/src/oskbd/linux.rs
@@ -13,6 +13,7 @@ use uinput_sys::uinput_user_dev;
 
 use crate::custom_action::*;
 use crate::keys::*;
+use libc::c_char;
 use libc::input_event as raw_event;
 
 // file i/o
@@ -88,10 +89,14 @@ impl KbdOut {
             }
 
             let mut uidev: uinput_user_dev = mem::zeroed();
-            uidev.name[0] = 'k' as i8;
-            uidev.name[1] = 't' as i8;
-            uidev.name[2] = 'r' as i8;
-            uidev.name[3] = 'l' as i8;
+
+            const PROG_NAME: &[u8] = "kanata".as_bytes();
+            let copy_len = std::cmp::min(PROG_NAME.len(), uidev.name.len());
+            assert!(uidev.name.len() >= copy_len);
+            for (i, c) in PROG_NAME.iter().copied().enumerate().take(copy_len) {
+                uidev.name[i] = c as c_char;
+            }
+
             uidev.id.bustype = 0x3; // BUS_USB
             uidev.id.vendor = 0x1;
             uidev.id.product = 0x1;

--- a/src/oskbd/linux.rs
+++ b/src/oskbd/linux.rs
@@ -92,7 +92,7 @@ impl KbdOut {
 
             const PROG_NAME: &[u8] = "kanata".as_bytes();
             let copy_len = std::cmp::min(PROG_NAME.len(), uidev.name.len());
-            assert!(uidev.name.len() >= copy_len);
+            assert!(copy_len <= uidev.name.len());
             for (i, c) in PROG_NAME.iter().copied().enumerate().take(copy_len) {
                 uidev.name[i] = c as c_char;
             }

